test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -229,4 +229,118 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
+				corev1.ResourceMemory: resource.MustParse("16Gi"),
+			}
+
+			By("Padding non selected nodes")
+			// TODO This should be calculated as 3/4 of requiredRes
+			paddingRes := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("12"),
+				corev1.ResourceMemory: resource.MustParse("12Gi"),
+			}
+
+			var paddingPods []*corev1.Pod
+			for _, nodeName := range nrtCandidateNames.List() {
+
+				nrtInfo, err := e2enrt.FindFromList(nrtCandidates, nodeName)
+				Expect(err).NotTo(HaveOccurred(), "missing NRT info for %q", nodeName)
+
+				baseload, err := nodes.GetLoad(fxt.K8sClient, nodeName)
+				Expect(err).NotTo(HaveOccurred(), "cannot get base load for %q", nodeName)
+
+				for idx, zone := range nrtInfo.Zones {
+					zoneRes := paddingRes.DeepCopy() // extra safety
+					if idx == 0 {                    // any zone is fine
+						baseload.Apply(zoneRes)
+					}
+
+					podName := fmt.Sprintf("padding%s-%d", nodeName, idx)
+					padPod, err := makePaddingPod(fxt.Namespace.Name, podName, zone, zoneRes)
+					Expect(err).NotTo(HaveOccurred(), "unable to create padding pod %q on zone", podName, zone.Name)
+
+					padPod, err = pinPodTo(padPod, nodeName, zone.Name)
+					Expect(err).NotTo(HaveOccurred(), "unable to pin pod %q to zone", podName, zone.Name)
+
+					err = fxt.Client.Create(context.TODO(), padPod)
+					Expect(err).NotTo(HaveOccurred(), "unable to create pod %q on zone", podName, zone.Name)
+
+					paddingPods = append(paddingPods, padPod)
+				}
+			}
+
+			failedPods := e2ewait.ForPodListAllRunning(fxt.Client, paddingPods)
+			for _, failedPod := range failedPods {
+				_ = objects.LogEventsForPod(fxt.K8sClient, failedPod.Namespace, failedPod.Name)
+			}
+			Expect(failedPods).To(BeEmpty(), "some padding pods have failed to run")
+		})
+
+		It("[test_id:47592][tier2][unsched] a daemonset with a guaranteed pod resources available on one node/one single numa zone but not in any other node", func() {
+
+			By("Scheduling the testing daemonset")
+			dsName := "test-ds"
+
+			podLabels := map[string]string{
+				"test": "test-daemonset",
+			}
+			nodeSelector := map[string]string{}
+			ds := objects.NewTestDaemonset(podLabels, nodeSelector, fxt.Namespace.Name, dsName, objects.PauseImage, []string{objects.PauseCommand}, []string{})
+			ds.Spec.Template.Spec.SchedulerName = serialconfig.Config.SchedulerName
+			ds.Spec.Template.Spec.Containers[0].Resources.Limits = requiredRes
+
+			err := fxt.Client.Create(context.TODO(), ds)
+			Expect(err).NotTo(HaveOccurred(), "unable to create deployment %q", ds.Name)
+
+			By(fmt.Sprintf("checking daemonset pods have been scheduled with the topology aware scheduler %q ", serialconfig.Config.SchedulerName))
+			pods, err := schedutils.ListPodsByDaemonset(fxt.Client, *ds)
+			Expect(err).ToNot(HaveOccurred(), "Unable to get pods from daemonset %q:  %v", ds.Name, err)
+
+			//TODO: should not we wait until pods have at least been scheduled? how?
```

Comment:
we need to wait for the DaemonSet to be ready. If the tests must succeed, the daemonset must go ready, or we can fail right at that pojt

Reply:
Yeah, my problem here was that our daemonset will hopefully never be "ready" as pods should be executed just on a subset of the desired nodes. 
Maybe, as we know the number of nodes with enough resources to accommodate daemonset pods, we could wait for a specific number of pods to be "Ready", as a special case for the `ForDaemonSetReady` function we already have in place.

Reply:
Unless the current approach is flaky, I think it's good enough and we can iterate/improve later. For extra points, we may want to file an issue to track this case.

Reply:
Oook ... sorry somehow I miss your second comment ... 
It could be flaky if the pod running in the node with enough resources reaches Running state faster then the other pods are scheduled into the other nodes and fail due to lack of resources, but if that case is possible (not sure) it seems very unlikely to me.